### PR TITLE
Update 1.10.0

### PR DESCRIPTION
**Pull request by neon60**
Update 1.10.0

### CODE REVIEW

**Comment by neon60** [comment]
Initiated the final internal review parallelly.

File: CMakeLists.txt [hunk 1]
```diff
@@ -33,7 +33,7 @@ endif()
 set(CMAKE_INSTALL_RPATH_USE_LINK_PATH TRUE CACHE BOOL "Add paths to linker search and installed rpath")
 
 # AMD targets
-set(AMDGPU_TARGETS gfx900:xnack-;gfx906:xnack-;gfx908:xnack- CACHE STRING "List of specific machine types for library to target")
+set(AMDGPU_TARGETS gfx900;gfx906;gfx908 CACHE STRING "List of specific machine types for library to target")
```
Comment (saadrahim) [inline CMakeLists.txt:36]: Need to leave the xnack flags on

**Review by saadrahim — CHANGES_REQUESTED** [review]
The AMD CHANGELOG.md needs to be updated to reflect that we are going to rocThrust 1.10 for the next rocm release.

File: CMakeLists.txt (hunk 1 above)
Comment (neon60) [reply CMakeLists.txt:36]: If we leave the xnack, I am getting the followinf error on rocm3.9
```
Invalid offload arch combinations: gfx906 and gfx906:xnack- 
```

Comment (saadrahim) [reply CMakeLists.txt:36]: Please see https://github.com/ROCmSoftwarePlatform/rocRAND/pull/158

**Review by saadrahim — APPROVED** [review]
Fine by me now.

File: CHANGELOG.md [hunk 2]
```diff
@@ -4,6 +4,7 @@ Full documentation for rocThrust is available at [https://rocthrust.readthedocs.
 
 ## [Unreleased rocThrust-2.10.7 for ROCm 4.0.0]
 ### Added
+- Updated to upstream Thrust 1.9.10
```
Comment (saadrahim) [inline CHANGELOG.md:7]: Should this be 1.10.0?

File: examples/cuda/custom_temporary_allocation.cu [hunk 3]
```diff
@@ -0,0 +1,184 @@
...
+// If a hit is found in the cache, we quickly return the cached allocation
+// instead of resorting to the more expensive thrust::cuda::malloc.
+
+// Note: Thrust now has its own caching allocator layer; if you just need a
+// caching allocator, you ought to use that. This example is still useful
+// as a demonstration of how to use a Thrust custom allocator.
+
+// Note: this implementation cached_allocator is not thread-safe. If multiple
+// (host) threads use the same cached_allocator then they should gain exclusive
+// access to the allocator before accessing its methods.
```
Comment (saadrahim) [inline examples/cuda/custom_temporary_allocation.cu:25]: Do these examples work on AMD hardware?

**Review by saadrahim — CHANGES_REQUESTED** [review]
Please answer the version number question.

File: examples/cuda/custom_temporary_allocation.cu (hunk 3 above)
Comment (neon60) [reply examples/cuda/custom_temporary_allocation.cu:25]: Currently this is not part of the build and we do not test this.

File: CHANGELOG.md (hunk 2 above)
Comment (neon60) [reply CHANGELOG.md]: Yes. Fixed.

File: examples/cuda/custom_temporary_allocation.cu (hunk 3 above)
Comment (neon60) [reply examples/cuda/custom_temporary_allocation.cu:25]: It's not in the build yet. Created a separate issue for this.

**Review by saadrahim — APPROVED** [review]
Thanks for the changes.

**Pull request merged**